addons/l10n_id_efaktur_coretax/tests/test_l10n_id_efaktur_coretax.py

```diff
@@ -0,0 +1,465 @@
+        """
+        1) contact with l10n_id_pkp with l10n_id_kode_transaksi=04
+        2) use 11% tax
+        """
+        super().setUpClass(chart_template_ref="l10n_id.l10n_id_chart")
+
+        cls.env.company.street = "test"
+        cls.env.company.phone = "12345"
+        cls.env.company.vat = "1234567890123456"
+
+        cls.partner_id = cls.env['res.partner'].create({"name": "l10ntest", "l10n_id_pkp": True, "l10n_id_kode_transaksi": "04", "vat": "1234567890123457"})
+        cls.tax_id = cls.env['account.tax'].create({"name": "test tax", "type_tax_use": "sale", "amount": 11.0})
+        cls.product_id = cls.env['product.product'].create({'name': 'Product A'})
+
+        path = "l10n_id_efaktur_coretax/tests/results/sample.xml"
+        with tools.file_open(path, mode='rb') as test_file:
+            cls.sample_xml = test_file.read()
+
+    def test_product_code_default(self):
+        """ Test interaction when changing the product type between 'consu' and 'service' which
+        will trigger change in `l10n_id_product_code` """
+
+        product = self.env['product.template'].create({
+            "name": "test product",
+            "type": "consu"
+        })
+        self.assertEqual(product.l10n_id_product_code, self.env.ref('l10n_id_efaktur_coretax.product_code_000000_goods'))
+
+        product.type = "service"
+        self.assertEqual(product.l10n_id_product_code, self.env.ref('l10n_id_efaktur_coretax.product_code_000000_service'))
+
+    # ==================================================
+    # Test conditions that dont allow efaktur generation
+    # ==================================================
+
+    def _verify_error_message(self, ex, err_count, messages=[]):
+        """ Verify that there are `err_count` number of errors and that all snippets of messages
+        exist in the error message
+        """
+        exception_msg = str(ex.exception)
+        actual_count = len(exception_msg.split('\n')) - 1  # -1 because the first line of error message ("Unable to Download ...") is always there
+
+        self.assertEqual(actual_count, err_count)
+        self.assertTrue(all(msg in exception_msg for msg in messages))
+
+    def test_download_efaktur_invalid_invoice(self):
+        """ Test to ensure conditions related to invoice are enforced when downloading E-Faktur
+        we will test it out
+        """
+
+        # Report it not having an invoice
+        vendor_bill = self.env['account.move'].create({
+            'move_type': 'in_invoice',
+            'partner_id': self.partner_id.id,
+            'invoice_date': '2019-05-01',
+            'date': '2019-05-01',
+            'invoice_line_ids': [
+                (0, 0, {'name': 'line1', 'price_unit': 100000, 'quantity': 1})
+            ],
+        })
+        vendor_bill.action_post()
+
+        with self.assertRaises(ValidationError) as val:
+            vendor_bill.download_efaktur()
+
+        self._verify_error_message(val, 1, ["not an invoice"])
```

Comment:
Can't you use `assertRaisesRegex` to match the exception message with a regex/string?